### PR TITLE
[SYNPY-1334] Revamp getting started docs

### PR DESCRIPTION
**Pull request by BryanFauble**
**Problem:**

1. The getting started tutorial covered a few use-cases, however, it did not do a great job at teaching the tool to someone.
2. Some information in the authentication.md doc could use updating
3. Additional context in several areas was needed

**Solution:**

1. I rewrote several of the initial pages you are dropped into on the python docs page
2. I added context copied from the Synapse doc pages https://help.synapse.org/docs/ here so this is a self-contained experience
3. I updated the flow of several pages and documents
4. I started all of the tutorials that I wanted to write about and fully wrote the initial `Project`, `Folder`, `File` tutorials. **I'd like feedback around the flow of these tutorials and what I might be able to improve**.
5. I added information into the `Further Reading` section of the doc pages that we can expand on the Models/Concepts of the system. This was specifically done so I can link to them from the Tutorial section and I didn't need to crowd the tutorials with additional context around these concepts.
6. I created a scripts folder in tutorials where we can house full tutorial scripts we cover in each tutorial section.

### CODE REVIEW

File: docs/tutorials/python/file.md [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+In this tutorial you will:
+
+1. Upload several files to Synapse
+1. Print stored attributes about your files
+1. List all Folders and Files within my project
+
+
+## Prerequisites
+* Make sure that you have completed the [Folder](./folder.md) tutorial.
+* The tutorial assumes you have a number of files ready to upload. If you do not, create a test or dummy file.
```
Comment (BryanFauble) [inline docs/tutorials/python/file.md:37]: Looking for input what the best way to mention this part is. Perhaps we should create a python gist that creates the above file structure?

Comment (thomasyu888) [reply docs/tutorials/python/file.md:37]: Hmm, we could provide some dummy files. There's probably a "fake" small biology dataset we can download.

Comment (BryanFauble) [reply docs/tutorials/python/file.md:37]: Good idea. I am going to include some of the sample files I was using during the tutorial. I just echo'd some sample data into files and named them

File: docs/tutorials/sample_files/my_ad_project/single_cell_RNAseq_batch_1/SRR12345678_R2.fastq.gz
Comment (thomasyu888) [inline docs/tutorials/sample_files/my_ad_project/single_cell_RNAseq_batch_1/SRR12345678_R2.fastq.gz:1]: Might be good to outline that these aren't actual fastq files

File: synapseutils/walk_functions.py [hunk 2]
```diff
@@ -26,6 +26,29 @@ def walk(
...
+                        f"Directory ({directory_name[1]}): {directory_path[0]}/{directory_name[0]}"
+                    )
+
+                for file in file_name:
+                    print(f"File ({file[1]}): {directory_path[0]}/{file[0]}")
+
+        The output will look like this:
+
+            Directory (syn12345678): My Project Name/my_directory_name
+            File (syn23456789): My Project Name/my_directory_name/fileA.txt
```
Comment (thomasyu888) [inline synapseutils/walk_functions.py:50]: Assuming that there is only one folder and one file in the directory

File: docs/tutorials/python/tutorial_scripts/file.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+syn = synapseclient.login()
+
+# Retrieve the project ID
+my_project_id = syn.findEntityId(
+    name="My uniquely named project about Alzheimer's Disease"
+)
+
+# Retrieve the IDs of the folders I want to upload to
+batch_1_folder = syn.findEntityId(
```
Comment (thomasyu888) [inline docs/tutorials/python/tutorial_scripts/file.py:18]: This is a nice use of `findEntityId`

I don't know if we want to outline that you can use the UI to find the synapse id

File: docs/tutorials/python/tutorial_scripts/annotation.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+# Retrieve the folders I want to annotate files in
+batch_1_folder_id = syn.findEntityId(
+    name="single_cell_RNAseq_batch_1", parent=my_project_id
+)
+
+print(f"Batch 1 Folder ID: {batch_1_folder_id}")
+
+
+# Define the annotations I want to set
+annotation_values = {
```
Comment (thomasyu888) [inline docs/tutorials/python/tutorial_scripts/annotation.py:25]: The annotation values can be set during the initial upload.  not sure if we wanted to complicate the upload script

File: docs/tutorials/authentication.md [hunk 5]
```diff
@@ -2,14 +2,78 @@
 
 There are multiple ways one can login to Synapse. We recommend users to choose the method that fits their workflow.
 
+## Prerequisites
+
+* Create a [personal access token](https://help.synapse.org/docs/Managing-Your-Account.2055405596.html#ManagingYourAccount-PersonalAccessTokens) token obtained from synapse.org under your Settings.
+    * Note that a token must minimally have the *view* scope to be used with the Synapse Python Client.
```
Comment (thomasyu888) [inline docs/tutorials/authentication.md:8]: Not sure if we wanted to outline that "view" scope won't be able to upload data into synapse (especially for the tutorials)

File: docs/explanations/domain_models_of_synapse.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
+# Domain Model of Synapse
+Here you'll find more information about how all of the models/concepts of Synapse
+work together and what their purpose is.
+
+## Entity
+Any distinct item in Synapse that has its own synID, including a Project, Folder, File, Wiki, Dataset, and View, among other items.
```
Comment (thomasyu888) [inline docs/explanations/domain_models_of_synapse.md:6]:
```suggestion
Any distinct item in Synapse that has its own Synapse Id, including a Project, Folder, File, Wiki, Dataset, and View, among other items.
```

File: docs/explanations/domain_models_of_synapse.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+Synapse entities are identified by a unique identifier called a Synapse ID, or synID. The synID is represented by the prefix “syn” followed by numbers (for example, syn12345678). This identifier can be used to refer to a specific entity on the web and through the programmatic clients.
+
+## [Projects](../tutorials/python/project.md)
+Projects in Synapse are “containers” that group relevant content and people together. All data must be uploaded into a project. Projects can be private so only you can see the contents, they can be shared with your collaborators, or they can be made public so anyone on the web can view your research.
```
Comment (thomasyu888) [inline docs/explanations/domain_models_of_synapse.md:11]:
```suggestion
Projects in Synapse are “containers” that group relevant content and people together. All data must be uploaded into a Project. Projects can be private so only you can see the contents, they can be shared with your collaborators, or they can be made public so anyone on the web can view your research.
```

File: docs/explanations/domain_models_of_synapse.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+Projects help you to:
+
+* **Organize your work**: With Synapse, you can organize the parts in your workflow (data, code, etc) into a hierarchy like a file system. With the tabs across the top of each project, you can easily navigate to your wikis, files and folders, SQL-based tables, and even conduct conversations using discussion forums.
```
Comment (thomasyu888) [inline docs/explanations/domain_models_of_synapse.md:15]: Does `Project` need to be capitalized throughout the document?

**Review by thomasyu888 — APPROVED** [review]
🔥 LGTM! I'm going to pre-approve because we can always continuously improve documentation.  

It will be continued iteration.

File: docs/tutorials/configuration.md [hunk 9]
```diff
@@ -1,6 +1,11 @@
 # Configuration
 
-The synapse python client can be configured either programmatically or by using a configuration file. When installing the Synapse Python client, the `.synapseConfig` is added to your home directory. This configuration file is used to store a number of configuration options, including your Synapse authtoken, cache, and multi-threading settings.
+The synapse python client can be configured either programmatically or by using a configuration file.
```
Comment (thomasyu888) [inline docs/tutorials/configuration.md:3]:
```suggestion
The Synapse Python client can be configured either programmatically or by using a configuration file.
```

File: docs/tutorials/sample_files/my_ad_project/single_cell_RNAseq_batch_1/SRR12345678_R2.fastq.gz
Comment (BryanFauble) [reply docs/tutorials/sample_files/my_ad_project/single_cell_RNAseq_batch_1/SRR12345678_R2.fastq.gz:1]: I added a readme to the root of the folder and added a note where this directory is linked from in the tutorials

File: synapseutils/walk_functions.py (hunk 2 above)
Comment (BryanFauble) [reply synapseutils/walk_functions.py:50]: Clarified this in the comment

File: docs/tutorials/python/tutorial_scripts/file.py (hunk 3 above)
Comment (BryanFauble) [reply docs/tutorials/python/tutorial_scripts/file.py:18]: For this I am going to focus solely on using this programmatic approach. I am only planning to mention the UI to look at results of doing something.

File: docs/tutorials/python/tutorial_scripts/annotation.py (hunk 4 above)
Comment (BryanFauble) [reply docs/tutorials/python/tutorial_scripts/annotation.py:25]: So in this script I am actually showing both ways. Uploading a file with annotations being set starts on L52.

File: docs/tutorials/authentication.md (hunk 5 above)
Comment (BryanFauble) [reply docs/tutorials/authentication.md:8]: Sure. Good idea. I'll mention this

File: docs/explanations/domain_models_of_synapse.md (hunk 7 above)
Comment (BryanFauble) [reply docs/explanations/domain_models_of_synapse.md:11]: I went the other way and lowercased things for consistency

File: docs/explanations/domain_models_of_synapse.md (hunk 8 above)
Comment (BryanFauble) [reply docs/explanations/domain_models_of_synapse.md:15]: I went the other way and lowercased things for consistency

**Pull request merged**